lib/services/sourced_track/sources/youtube.dart

```diff
@@ -79,14 +79,17 @@ class YoutubeSourcedTrack extends SourcedTrack {
   }
 
   static SourceMap toSourceMap(StreamManifest manifest) {
-    final m4a = manifest.audioOnly
+    var m4a = manifest.audioOnly
         .where((audio) => audio.codec.mimeType == "audio/mp4")
         .sortByBitrate();
 
-    final weba = manifest.audioOnly
+    var weba = manifest.audioOnly
         .where((audio) => audio.codec.mimeType == "audio/webm")
         .sortByBitrate();
 
+    m4a = m4a.isEmpty ? weba : m4a;
+    weba = weba.isEmpty ? m4a : weba;
```

Comment:
Please create a new `List` when swapping values for variables. Otherwise, it can lead to unwanted and hard to debug issues.

```suggestion
    m4a = m4a.isEmpty ? weba.toList() : m4a;
    weba = weba.isEmpty ? m4a.toList() : weba;
```